hokusai/cli/base.py

```diff
@@ -24,18 +24,16 @@ def console():
 
 
 @base.command(context_settings=CONTEXT_SETTINGS)
-@click.option('--kubectl-version', type=click.STRING, required=False, help='The version of kubectl to install')
-@click.option('--s3-bucket', type=click.STRING, required=False, help="The S3 bucket name containing your org's kubectl config file")
-@click.option('--s3-key', type=click.STRING, required=False, help="The S3 key of your org's kubectl config file")
-@click.option('--config_file', type=click.STRING, required=False, help="Your org's kubectl config file")
-@click.option('--platform', type=click.Choice(['darwin', 'linux']), default='darwin', help='The platform OS (default: darwin)')
-@click.option('--install-to', type=click.STRING, default='/usr/local/bin', help='Install kubectl to (default: /usr/local/bin)')
-@click.option('--install-config-to', type=click.STRING, default=os.path.join(os.environ.get('HOME'), '.kube'), help='Install kubectl config to (default: ~/.kube)')
-@click.option('-v', '--verbose', type=click.BOOL, is_flag=True, help='Verbose output')
-def configure(kubectl_version, s3_bucket, s3_key, config_file, platform, install_to, install_config_to, verbose):
-  """Install and configure kubectl"""
+@click.option('--kubeconfig-dir', type=click.STRING, default=None, help='Directory to install kubeconfig into. The kubeconfig file will be named "config". (default: None)')
+@click.option('--kubectl-dir', type=click.STRING, default=None, help='Directory to install kubectl into. The executable will be named "kubectl". (default: None)')
+@click.option('--new-config', type=click.STRING, default=None, help='Path to new Hokusai global config. Can be AWS S3 location (e.g. s3://bucket/prefix/to/file), or local file (e.g. /path/to/file, ./path/to/file). New config will overwrite existing ~/.hokusai.yml')
```

Comment:
This naming is a little unclear to me. It's like a config "source" or "import," basically, right? I will browse some other CLIs' options to refine an opinion about this...

Reply:
Actually, having read through this more fully now, I think the closest analog is `kubeconfig-source-uri`. Why not name this `source-uri`? The case of importing from a local file could maybe still be supported, but that scenario seems odd to me so I'm curious how you think it would be used.

Reply:
`source-uri` sounds ambiguous but the help string does elaborate `Path to new Hokusai global config...`, so I suppose it's fine. Allowing local file source is so that one is not locked into using S3.

Reply:
Alas, this command line option does not work. The way we [load all modules upfront](https://github.com/artsy/hokusai/blob/09bcc2fc76249af077e6230c697d1232211ff2fa/hokusai/__init__.py#L4) when running any Hokusai command, Kubectl service gets loaded too, and it [instantiates global config object](https://github.com/artsy/hokusai/pull/379/files#diff-a4ddf5adce9f1ad6808af8d4aa9dcaa680aebc2d4fa1ba58c453b245602f32ccR10) at module level. That instantiation is immediately done when the module is imported. The result is that when any Hokusai command is run, `~/.hokusai.yml` must already exist, so user won't get a chance to run `hokusai configure --new-config`.

This blanket upfront loading of modules has been causing problem again and again. I am not sure what to do about it at the moment.

For now, we can just [replace the command line option with an Env var](https://github.com/artsy/hokusai/pull/379/commits/60f16a31d33dd2de0c66521c80d2b8398334a01c) which takes precedence over `~/.hokusai.yml`, for config read. Config write will continue to go to `~/.hokusai.yml`.

Reply:
The error I got was:
```
artsy:hokusai-sandbox jxu$ /Users/jxu/.pyenv/versions/hokusai/bin/hokusai configure --help
Error: failed to copy /Users/jxu/.hokusai.yml to /var/folders/n6/mjsp4zt94yn60gfn7dzr248w0000gn/T/tmpkd40shy_/hokusai.yml
Error: Not able to load config from /Users/jxu/.hokusai.yml
Traceback (most recent call last):
  File "/Users/jxu/.pyenv/versions/hokusai/bin/hokusai", line 3, in <module>
    from hokusai.command_line import main
  File "/Users/jxu/code/hokusai/hokusai/__init__.py", line 4, in <module>
    from hokusai.commands import *
  File "/Users/jxu/code/hokusai/hokusai/commands/__init__.py", line 2, in <module>
    from hokusai.commands.check import check
  File "/Users/jxu/code/hokusai/hokusai/commands/check.py", line 12, in <module>
    from hokusai.services.kubectl import Kubectl
  File "/Users/jxu/code/hokusai/hokusai/services/kubectl.py", line 10, in <module>
    global_config = HokusaiGlobalConfig()
  File "/Users/jxu/code/hokusai/hokusai/lib/global_config.py", line 15, in __init__
    self._config = ConfigLoader(uri).load()
  File "/Users/jxu/code/hokusai/hokusai/lib/config_loader.py", line 32, in load
    uri_to_local(self.uri, tmpdir, filename)
  File "/Users/jxu/code/hokusai/hokusai/lib/common.py", line 197, in uri_to_local
    local_to_local(parsed_uri.path, target_dir, target_file)
  File "/Users/jxu/code/hokusai/hokusai/lib/common.py", line 97, in local_to_local
    shutil.copyfile(source_path, target_path)
  File "/Users/jxu/.pyenv/versions/3.9.10/lib/python3.9/shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/jxu/.hokusai.yml'
```